compiler-cli/src/build_lock.rs

```diff
@@ -8,34 +8,59 @@ use gleam_core::{
 use strum::IntoEnumIterator;
 
 #[derive(Debug)]
-pub(crate) struct BuildLock {
-    directory: Utf8PathBuf,
+pub(crate) enum BuildLock {
```

Comment:
Is it ok for it to be enum? Or we can construct `lock_filename` right in the constructor

Reply:
Making the path in the constructor sounds good to me